### DISCUSSION TITLE
Where is it determined which promo / adjustment "wins"?

### DISCUSSION THREAD

**Discussion by hrdchz** [solidusio/solidus#3972] [Troubleshooting] [answered]
On our store it is possible to have multiple promos running at the same time. For example:

```
Promo A:
- code: xyz
- discount: 10% off order

Promo B:
- code: abc
- discount: 20% off order

```

So on an order with both codes activated, the discount from Promo B is applied.

What decides the "winning" adjustment?

EDIT: removed confusing "Promo C" from scenario.

**Comment by jarednorman** [comment, marked as answer]
You should only be getting whichever one applies the largest discount. 🤔

**Reply by hrdchz** [reply]
Ah ok, I think my question is more like, regardless of the line item also showing up, what is deciding the A vs B discount?

I think Promo A and B here are handled by PromotionHandler::Coupon but C is handled by PromotionHandler::Cart. So I'm guessing those are different kinds of adjustments. I'm expecting to find some AdjustmentDecider class but I can't seem to grep it out.

**Reply by jarednorman** [reply]
I think you're looking for this code: https://github.com/solidusio/solidus/blob/master/core/app/models/spree/order_updater.rb#L102-L118

**Reply by jarednorman** [reply]
There is a "promotion chooser" once you dig in: https://github.com/solidusio/solidus/blob/93bfdb4ab11951b4bccc3d8a42f9bde3b77205ac/core/app/models/spree/order_updater.rb#L203

**Reply by jarednorman** [reply]
This is the default one: https://github.com/solidusio/solidus/blob/a17109b07a817ae34bc61491c3ed1ea824ee32c5/core/app/models/spree/promotion_chooser.rb

**Reply by hrdchz** [reply]
ah ha yes! PromotionChooser, not AdjustmentDecider.
